text/0016-standard-modes.md

```diff
@@ -0,0 +1,295 @@
+
+More specific flight behaviour/modes are defined in _custom modes_, which are specific to each flight stack.
+Mechanisms are provided to set the base and custom modes ([MAV_CMD_DO_SET_MODE](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_SET_MODE)).
+The `HEARTBEAT` contains the current/active `base_mode` and `custom_mode`.
+There are no mechanisms query the available custom modes.
+
+Most autopilots implement a set of custom flight modes that have very similar behaviour or intent, for example: position-hold mode, altitude-hold mode, mission mode, safety return/RTL.
+
+However because these are all identified by different custom mode identifiers on different flight stacks, there is no way to be sure what these "mean" without pre-existing knowledge.
+
+
+# Detailed Design
+
+The design proposes a small set of modes that are (broadly speaking) commonly implemented across flight stacks, along with mechanisms to:
+- determine what modes are available
+- set the standard mode
+- infer the current standard mode, if active.
+
+
+## Standard modes
+
+Standard modes are those that are implemented in a similar way by most flight stacks, and which a GCS can present as more or less the same thing for any flight stack.
+
+The precise mechanics and behaviour do not have to be identical, but the broad intent has to be similar, and any differing behaviour across flight statcks should not be unexpected by a user.
+
+A flight stack is not _required_ to support all of these modes.
+
+The standard modes will be defined in an enum `MAV_STANDARD_MODE`, which has an initial value `0` is an enum that indicates "not a standard mode".
+
+The proposed initial set of modes is:
+```xml
+    <enum name="MAV_STANDARD_MODE">
+      <description>Standard modes with a well understood meaning across flight stacks and vehicle types.
+        For example, most flight stack have the concept of a "return" or "RTL" mode that takes a vehicle to safety, even though the precise mechanics of this mode may differ.
+        Modes may be set using MAV_CMD_DO_SET_STANDARD_MODE.
+      </description>
+      <entry value="0" name="MAV_STANDARD_MODE_NON_STANDARD">
+        <description>Non standard mode.
+          This may be used when reporting the mode if the current flight mode is not a standard mode.
+        </description>
+      </entry>
+      <entry value="1" name="MAV_STANDARD_MODE_POSITION_HOLD">
+        <description>Position mode (manual).
+          Position-controlled and stabilized manual mode.
+          When sticks are released vehicles return to their level-flight orientation.
+          Multicopter (MC) vehicles actively brake and hold both position and altitude against wind and external forces.
+          Fixed-wing (FW) vehicles maintain current track and altitude against wind and external forces.
+          Hybrid MC/FW  ("VTOL") vehicles behave according to their current configuration/mode (FW or MC).
```

Comment:
Still disagree with this description.  "Circle fixed point" is a better description/behaviour for "position hold" in FW

Reply:
Bugger, of course you're right. Off to get a coffee. I will fix.

Reply:
@auturgy Actually no, this is what I meant:

On PX4 [this is what the mode](http://docs.px4.io/master/en/flight_modes/position_fw.html) does on FW - it is a stabilized mode when flying (i.e. roll/pitch limited) but when you release the sticks it stabilizes orientation to the earth plane and holds the track - ie if you draw an imaginary line to infinity and get blown of course you return to that line. 

I'm trying to work out what ardupilot does but I can't tell which is the most comparable flight mode from these descriptions - thoughts: https://ardupilot.org/plane/docs/flight-modes.html ?

I can certainly accept that position hold would be more logical as a circle/loiter (rather than holding track) based on the name. Not sure if it is "safer" - depends on whether you're more happy to take back over flying from straight flight or perhaps you'd just prefer it to circle while you think about it.

Well worth discussing.

Reply:
"Cruise" is the closest ArduPilot mode to what you describe.  Perhaps omit Position Hold for fixed-wing.  Acknowledging PX4 are using it to describe a mode, it actually doesn't make any sense as a mode name for fixed wing aircraft (that can't hover).

Reply:
A quick check of iNav (https://github.com/iNavFlight/inav/wiki/Navigation-modes and https://github.com/iNavFlight/inav/wiki/Modes) shows that they call this behaviour "NAV CRUISE", with "NAV POSHOLD" being loiter in circles around a point.